src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -122,15 +130,34 @@ class Config:
     )
 
 
+class GeneratedText(BaseModel):
+    text: str = Field(
+        description="The generated sequence for a given prompt. If "
+        "streaming is enabled, this will be the next generated token."
+    )
+    score: Optional[Any] = Field(
+        description="The score for the generated token or sequence. "
+        "The scores have the shape [sequence_length, vocab_size]"
+    )
+    finished: bool = Field(description="Whether generation has stopped.")
+    finished_reason: str = Field(
+        description="The reason for generation to stop. "
+        "Defined by FinishReason. One of stop, length, or time."
+    )
```

Comment:
Tempted to think that `finished_reason` could just be an `Optional[str]` so we know it is unfinished if `None`, but happy either way

Reply:
Yup - there’s a draft PR to enable streaming where I update this to be optional